kerbauth_test.go

```diff
@@ -0,0 +1,149 @@
+package mssql
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/jcmturner/gokrb5/v8/client"
+	"github.com/jcmturner/gokrb5/v8/config"
+	"github.com/jcmturner/gokrb5/v8/credentials"
+	"github.com/jcmturner/gokrb5/v8/keytab"
+)
+
+func TestGetKRB5Auth(t *testing.T) {
+	krbConf := &config.Config{}
+	krbKeytab := &keytab.Keytab{}
+	krbCache := &credentials.CCache{}
+
+	got, _ := getKRB5Auth("", "MSSQLSvc/mssql.domain.com:1433", krbConf, krbKeytab, krbCache)
+	keytab := &krb5Auth{username: "",
+		realm:      "domain.com",
+		serverSPN:  "MSSQLSvc/mssql.domain.com:1433",
+		port:       1433,
+		krb5Config: krbConf,
+		krbKeytab:  krbKeytab,
+		krbCache:   krbCache,
+		state:      0}
+
+	res := reflect.DeepEqual(got, keytab)
+	if !res {
+		t.Errorf("Failed to get correct krb5Auth object\nExpected:%v\nRecieved:%v", keytab, got)
+	}
+
+	got, _ = getKRB5Auth("", "MSSQLSvc/mssql.domain.com:1433", krbConf, krbKeytab, krbCache)
+	keytab = &krb5Auth{username: "",
+		realm:      "domain.com",
+		serverSPN:  "MSSQLSvc/mssql.domain.com:1433",
+		port:       1433,
+		krb5Config: krbConf,
+		krbKeytab:  krbKeytab,
+		krbCache:   krbCache,
+		state:      0}
+
+	res = reflect.DeepEqual(got, keytab)
+	if !res {
+		t.Errorf("Failed to get correct krb5Auth object\nExpected:%v\nRecieved:%v", keytab, got)
+	}
+
+	_, val := getKRB5Auth("", "MSSQLSvc/mssql.domain.com", krbConf, krbKeytab, krbCache)
+	if val {
+		t.Errorf("Failed to get correct krb5Auth object: no port defined")
+	}
+
+	got, _ = getKRB5Auth("", "MSSQLSvc/mssql.domain.com:1433@DOMAIN.COM", krbConf, krbKeytab, krbCache)
+	keytab = &krb5Auth{username: "",
+		realm:      "DOMAIN.COM",
+		serverSPN:  "MSSQLSvc/mssql.domain.com:1433",
+		port:       1433,
+		krb5Config: krbConf,
+		krbKeytab:  krbKeytab,
+		krbCache:   krbCache,
+		state:      0}
+
+	res = reflect.DeepEqual(got, keytab)
+	if !res {
+		t.Errorf("Failed to get correct krb5Auth object\nExpected:%v\nRecieved:%v", keytab, got)
+	}
+
+	_, val = getKRB5Auth("", "MSSQLSvc/mssql.domain.com:1433@domain.com@test", krbConf, krbKeytab, krbCache)
+	if val {
+		t.Errorf("Failed to get correct krb5Auth object due to incorrect serverSPN name")
+	}
+
+	_, val = getKRB5Auth("", "MSSQLSvc/mssql.domain.com:port@domain.com", krbConf, krbKeytab, krbCache)
+	if val {
+		t.Errorf("Failed to get correct krb5Auth object due to incorrect port")
+	}
+
+	_, val = getKRB5Auth("", "MSSQLSvc/mssql.domain.com:port", krbConf, krbKeytab, krbCache)
+	if val {
+		t.Errorf("Failed to get correct krb5Auth object due to incorrect port")
+	}
+}
+
+func TestInitialBytes(t *testing.T) {
+	krbConf := &config.Config{}
+	krbKeytab := &keytab.Keytab{}
+	krbCache := &credentials.CCache{}
+	krbObj := &krb5Auth{username: "",
+		realm:      "domain.com",
+		serverSPN:  "MSSQLSvc/mssql.domain.com:1433",
+		port:       1433,
+		krb5Config: krbConf,
+		krbKeytab:  krbKeytab,
+		krbCache:   krbCache,
+		state:      0,
+	}
+
+	_, err := krbObj.InitialBytes()
+	if err == nil {
```

Comment:
If the test is expecting a nil err please change the `t.ErrorF` messages to indicate that. Otherwise the code reads like `err == nil` should be changed to `err != nil`